File: vitefrontend/src/api/reports.ts
import { apiRequest } from './client'
import type { CategoryTotal } from '../types/domain'

export type DailyReport = {
  date: string
  totalMinutes: number
  byCategory: CategoryTotal[]
}

export type MonthlyReport = {
  month: string
  totalMinutes: number
  byCategory: CategoryTotal[]
  byDay: Array<{ date: string; totalMinutes: number }>
}

export function getDailyReport(date: string) {
  return apiRequest<DailyReport>(`/reports/daily?date=${date}`)
}

export function getMonthlyReport(month: string) {
  return apiRequest<MonthlyReport>(`/reports/monthly?month=${month}`)
}

File: fastapiBackend/app/crud/reports.py
def empty_daily_report(date: str) -> dict[str, object]:
    return {"date": date, "totalMinutes": 0, "byCategory": []}


def empty_monthly_report(month: str) -> dict[str, object]:
    return {"month": month, "totalMinutes": 0, "byCategory": [], "byDay": []}
